src/boards.ts

```diff
@@ -154,6 +157,25 @@ export class Board {
     this.controller.execute(command, data.username);
     callback(command.commandId);
   }
+  handleStartText(data: StartTextEvent, callback: StartAck) {
+    const text = new Text(
+      data.position,
+      ""
+    );
+    const command = new TextCommand(this.currentCommandId++, text, data.username);
+
+    this.controller.execute(command, data.username);
+    callback(command.commandId);
+  }
+
+  handleDoText(data: DoTextEvent) {
+    if (!this.controller.stack.get(data.commandId)) return;
+    const textCommand = this.controller.stack.get(
+        data.commandId,
+    )! as TextCommand;
+    textCommand.text.content = data.content;
+    textCommand.execute(this.namespace);
+  }
```

Comment:
```suggestion
  handleDoText(data: DoTextEvent) {
    if (!this.controller.stack.has(data.commandId)) return;
    const textCommand = this.controller.stack.get(
        data.commandId,
    )! as TextCommand;
    textCommand.text.content = data.content;
    textCommand.execute(this.namespace);
  }
```
use has attribute of map to check if it exists